read-fonts/src/collections/int_set/bitset.rs

```diff
@@ -614,6 +614,45 @@ impl Extend<u32> for U32Set {
     }
 }
 
+/// This helper accelerates set membership checks by caching the last page found.
+///
+/// If membership checks are in sorted order, or are typically clustered together then
+/// this will yield performance improvements over the standard U32Set::contains() call.
+pub struct U32SetContainsAccelerator<'a> {
+    set: &'a U32Set,
+    last_page_index: usize,
+    last_major_value: u32,
```

Comment:
In HB we just keep the last_page_index and lookup the major. Not sure how much overhead that has.

Reply:
It's not too much, only about a 5% difference and by not storing the major we can support the set changing since the cached index was stored so going to follow the hb approach.